db/friend.go

```diff
@@ -6,33 +6,33 @@ import (
 )
 
 type Friend struct {
-	DisplayName, MAC, IP string
+	DisplayName, MAC, IP, Username string
 }
 
 // Checks if the two users are friends given friend's display name
 func AreFriendsName(username string, friendDisplayName string) bool {
-	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
+	query := fmt.Sprintf("SELECT friend_display_name, friend_mac_address, friend_ip_address, friend_username FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
 	friends := ExecuteFriendsQuery(query)
 	return len(friends) > 0
 }
 
 // Checks if the two users are friends given friend's MAC Address
 func AreFriendsMac(username string, friendMacAddress string) bool {
-	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_mac_address=\"%s\"", friendsTableName, username, friendMacAddress)
+	query := fmt.Sprintf("SELECT friend_display_name, friend_mac_address, friend_ip_address, friend_username FROM %s WHERE username=\"%s\" AND friend_mac_address=\"%s\"", friendsTableName, username, friendMacAddress)
 	friends := ExecuteFriendsQuery(query)
 	return len(friends) > 0
 }
 
 // Add friend to the user's friend's table
-func AddFriend(username string, displayName string, macAddress string, ipAddress string) bool {
+func AddFriend(username string, displayName string, macAddress string, ipAddress string, friendUsername string) bool {
```

Comment:
This might be late to change the paradigm now, but wouldn't it make more sense if these methods belonged to a User?

This could be invoked like:

`u.AddFriend(f)` where f is a Friend struct, then you wouldn't have to pass each field of the friend struct here. It would make it a lot easier when it's called throughout, since we will be working with the User struct anyways.

Reply:
That is tough - DB functions that access a certain table should be grouped together. However at the same time the user object could use a function like that. I think its a matter of preference - we had decided to put it here originally, but I can move that if we decide its better.

Reply:
It's not really moving it. It's in the same package still, just in a different file, and it would belong to a struct instead of a class method. I think it's important to think about how they will be called - we will be working with structs, not primitive types.

Reply:
It is still moving it, even though it will remain in the same package. Like I said, both ways make sense, its more of how it will be called (which currently we don't have implemented), which will determine where it goes. If it is going to be called with the user struct, then it should be moved. But that needs to be figured out.

Reply:
We discussed offline - we will leave it the way it is.

Eventually, the methods can be camel cased since they will be package private, and then the User and Friend structs can have methods that invoke these db methods.